File: Part2/Vulnerable_Code3/python/des_ecb.py
from Crypto.Cipher import DES

# Remove or load from a secure key management system.
# Replace DES with AES (256-bit). For example, in CBC mode:
Crypto.Cipher.AES.new($ARGS, Crypto.Cipher.AES.MODE_CBC)
# Replace DES with AES (256-bit). For example, in CBC mode:
Crypto.Cipher.AES.new($ARGS, Crypto.Cipher.AES.MODE_CBC)
# Replace DES with AES (256-bit). For example, in CBC mode:
Crypto.Cipher.AES.new($ARGS, Crypto.Cipher.AES.MODE_CBC)
# Replace DES with AES (256-bit). For example, in CBC mode:
Crypto.Cipher.AES.new($ARGS, Crypto.Cipher.AES.MODE_CBC)
# Replace DES with AES (256-bit). For example, in CBC mode:
Crypto.Cipher.AES.new($ARGS, Crypto.Cipher.AES.MODE_CBC)
# Replace DES with AES (256-bit). For example, in CBC mode:
Crypto.Cipher.AES.new($ARGS, Crypto.Cipher.AES.MODE_CBC)
# Replace DES with AES (256-bit). For example, in CBC mode:
Crypto.Cipher.AES.new($ARGS, Crypto.Cipher.AES.MODE_CBC)
# Replace DES with AES (256-bit). For example, in CBC mode:
Crypto.Cipher.AES.new($ARGS, Crypto.Cipher.AES.MODE_CBC)
# Replace DES with AES (256-bit). For example, in CBC mode:
Crypto.Cipher.AES.new($ARGS, Crypto.Cipher.AES.MODE_CBC)
# Replace DES with AES (256-bit). For example, in CBC mode:
Crypto.Cipher.AES.new($ARGS, Crypto.Cipher.AES.MODE_CBC)
# Replace DES with AES (256-bit). For example, in CBC mode:
Crypto.Cipher.AES.new($ARGS, Crypto.Cipher.AES.MODE_CBC)
# Replace DES with AES (256-bit). For example, in CBC mode:
Crypto.Cipher.AES.new($ARGS, Crypto.Cipher.AES.MODE_CBC)
Crypto.Cipher.AES.new($ARGS, Crypto.Cipher.AES.MODE_CBC)

plaintext = b'This is a secret!'  # Input text
# Padding
pad_len = 8 - len(plaintext) % 8
plaintext += bytes([pad_len]) * pad_len

# Encrypt
ciphertext = cipher.encrypt(plaintext)
print("Encrypted:", ciphertext)

# Decrypt
decrypted = cipher.decrypt(ciphertext)
pad_len = decrypted[-1]
decrypted = decrypted[:-pad_len]
print("Decrypted:", decrypted)

File: Part2/Vulnerable_Code3/c/des_ecb.c
#include <openssl/des.h>
#include <stdio.h>
#include <string.h>

int main() {
    DES_cblock key = {"12345678"};
    DES_key_schedule schedule;
// Replace with AES key setup call, e.g.:
AES_set_encrypt_key(..., 256, &aes_key);
AES_set_encrypt_key(..., 256, &aes_key);
AES_set_encrypt_key(..., 256, &aes_key);
AES_set_encrypt_key(..., 256, &aes_key);
AES_set_encrypt_key(..., 256, &aes_key);
AES_set_encrypt_key(..., 256, &aes_key);
AES_set_encrypt_key(..., 256, &aes_key);
AES_set_encrypt_key(..., 256, &aes_key);
AES_set_encrypt_key(..., 256, &aes_key);
AES_set_encrypt_key(..., 256, &aes_key);
AES_set_encrypt_key(..., 256, &aes_key);

    const char *plaintext = "This is a secret!";
    char ciphertext[64] = {0};
    char decrypted[64] = {0};

    DES_ecb_encrypt((const_DES_cblock *)plaintext, (DES_cblock *)ciphertext, &schedule, DES_ENCRYPT);
    printf("Encrypted: %s\n", ciphertext);

    DES_ecb_encrypt((const_DES_cblock *)ciphertext, (DES_cblock *)decrypted, &schedule, DES_DECRYPT);
    printf("Decrypted: %s\n", decrypted);

    return 0;
}